deeplake/core/chunk_engine.py

```diff
@@ -809,26 +809,27 @@ def _samples_to_chunks(
         incoming_num_samples = len(samples)
         enc_ids: List[Optional[str]] = []
         enc_count = [0]
-        if extending:
-            if self.tensor_meta.htype == "text" and (
-                self.chunk_class != SampleCompressedChunk
-            ):
-                lengths = np.zeros(len(samples), dtype=np.uint32)
-                for i, s in enumerate(samples):
-                    try:
-                        s = s.numpy()
-                    except AttributeError:
-                        pass
+        if (
+            extending
+            and self.tensor_meta.htype == "text"
+            and (self.chunk_class != SampleCompressedChunk)
+        ):
+            lengths = np.zeros(len(samples), dtype=np.uint32)
+            for i, s in enumerate(samples):
+                try:
+                    s = s.numpy()
+                except AttributeError:
+                    pass
+                try:
+                    if s.dtype.name.startswith("str"):
+                        lengths[i] = len(str(s.reshape(())))
+                except AttributeError:
                     try:
-                        if s.dtype.name[:3] == "str":
-                            lengths[i] = len(str(s.reshape(())))
-                    except AttributeError:
-                        try:
-                            lengths[i] = s.__len__()
-                        except AttributeError:  # None
-                            lengths[i] = 0
-                        except TypeError:  # Numpy scalar str
-                            lengths[i] = str(s).__len__()
+                        lengths[i] = s.__len__()
+                    except AttributeError:  # None
+                        lengths[i] = 0
+                    except TypeError:  # Numpy scalar str
+                        lengths[i] = str(s).__len__()
```

Comment:
This is very hard to read change, I know it used to be even worth but can we introduce new methods instead:


```suggestion
    def _calculate_lengths(self, samples):
        """
        Calculates the lengths of the given samples.

        Args:
            samples (list): The list of samples to calculate lengths for.

        Returns:
            numpy.ndarray: An array of calculated lengths.
        """
        lengths = np.zeros(len(samples), dtype=np.uint32)
        for i, sample in enumerate(samples):
            lengths[i] = self._get_sample_length(sample)
        return lengths
        
   def _get_sample_length(self, sample):
      """
      Get the length of a single sample.

      Args:
          sample: A single sample whose length is to be calculated.

      Returns:
          int: The length of the sample.
      """
      # Convert to numpy array if possible
      if hasattr(sample, 'numpy'):
          sample = sample.numpy()

      # Handle numpy array with string data type
      if isinstance(sample, np.ndarray) and sample.dtype.name.startswith("str"):
          return len(str(sample.reshape(())))

      # Fallback for other types
      return len(str(sample)) if sample is not None else 0


     if (self.extending and self.tensor_meta.htype == "text" and self.chunk_class != SampleCompressedChunk):
         lengths = self._calculate_lengths(samples) 
```
    
    then apply these two methods instead. Haven't verified fully so, it might be producing not covering all of the edge cases, but you got the point

Reply:
To be honest I do not want to make my small cleanup initiative whole chunk_engine file rewrite